src/request/sign-multisig-transaction/SignMultisigTransactionApi.js

```diff
@@ -0,0 +1,284 @@
+                if (!(key instanceof Uint8Array)) throw new Error('not a Uint8Array');
+                publicKeys.push(new Nimiq.PublicKey(key));
+            }
+        } catch (error) {
+            const errorMessage = error instanceof Error ? error.message : String(error);
+            throw new Errors.InvalidRequestError(`Invalid public keys: ${errorMessage}`);
+        }
+
+        /** @type {{publicKey: Nimiq.PublicKey, commitments: Nimiq.Commitment[]}[]} */
+        const signers = [];
+        try {
+            if (!('signers' in object)) throw new Error('missing');
+            if (!Array.isArray(object.signers)) throw new Error('not an array');
+            for (const obj of object.signers) {
+                if (typeof obj !== 'object' || obj === null) throw new Error('not objects');
+
+                if (!('publicKey' in obj)) throw new Error('missing publicKey');
+                if (!(obj.publicKey instanceof Uint8Array)) throw new Error('publicKey not a Uint8Array');
+                const signerPublicKey = new Nimiq.PublicKey(obj.publicKey);
+                // Verify key is included in publicKeys as well
+                if (!publicKeys.find(publicKey => publicKey.equals(signerPublicKey))) {
+                    throw new Errors.InvalidRequestError('not in public keys');
+                }
+
+                if (!('commitments' in obj)) throw new Error('missing commitments');
+                if (!Array.isArray(obj.commitments) || obj.commitments.length < 2) {
+                    throw new Error('commitments must be an array with at least 2 elements');
+                }
+                /** @type {Nimiq.Commitment[]} */
+                const signerCommitments = [];
+                for (const commitment of obj.commitments) {
+                    if (!(commitment instanceof Uint8Array)) throw new Error('commitment must be a Uint8Array');
+                    signerCommitments.push(new Nimiq.Commitment(commitment));
+                }
+
+                signers.push({
+                    publicKey: signerPublicKey,
+                    commitments: signerCommitments,
+                });
+            }
+        } catch (error) {
+            const errorMessage = error instanceof Error ? error.message : String(error);
+            throw new Errors.InvalidRequestError(`Invalid signers: ${errorMessage}`);
+        }
+
+        /** @type {MultisigConfig['secrets']} */
+        let secrets;
+        if (Array.isArray(object.secrets)) {
+            secrets = object.secrets.map(
+                /**
+                 * @param {any} secret
+                 * @returns {Nimiq.RandomSecret}
+                 */
+                secret => {
+                    if (!(secret instanceof Uint8Array)) {
+                        throw new Errors.InvalidRequestError('Invalid secrets: must be Uint8Arrays');
+                    }
+                    return new Nimiq.RandomSecret(secret);
+                },
+            );
```

Comment:
If `secret` cannot be deserialized as a valid `RandomSecret`, the error is not mapped to a`InvalidRequestError`.
```suggestion
            try {
                for (const secret of object.secrets) {
                    if (!(secret instanceof Uint8Array)) throw new Error('must be Uint8Arrays');
                    secrets.push(new Nimiq.RandomSecret(secret));
                }
            } catch (error) {
                const errorMessage = error instanceof Error ? error.message : String(error);
                throw new Errors.InvalidRequestError(`Invalid secrets: ${errorMessage}`);
            }
```

Reply:
Resolved in #524